File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-2_Gestion-Employes/Implementation-Java/Caissiere.java
public class Caissiere extends Employe {
    public Caissiere(String numeroSecuriteSociale, String nom, double salaire, Employe superieur) {
        super(numeroSecuriteSociale, nom, salaire, "Caissiere", superieur);
    }

    @Override
    public double calculerSalaire() {
        return salaire;
    }

    @Override
    public String afficherSubordonnes() {
        return super.afficherSubordonnes() + " Aucuns";
    }
}

File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-2_Gestion-Employes/Implementation-Cpp/Caissiere.cpp
#include "Caissiere.h"

Caissiere::Caissiere(string numeroSecuriteSociale, string nom, double salaire, Employe * superieur) : Employe(numeroSecuriteSociale, nom, salaire, "Caissiere", superieur) {}

double Caissiere::calculerSalaire() const {
    return salaire;
}

string Caissiere::afficherSubordonnes() const {
    return Employe::afficherSubordonnes() + " Aucun";
}